src/lib/structures/expr.ml

```diff
@@ -1015,14 +1023,24 @@ let mk_ite cond th el =
     if ty == Ty.Tbool then mk_if cond th el
     else mk_term (Sy.Op Sy.Tite) [cond; th; el] ty
 
-let [@inline always] const_term e =
-  (* we use this function because depth is currently not correct to
+let rec is_const_term e =
```

Comment:
I thought this was replacing the pre-existing `const_term` function. Given that it's not, it is very confusing to have both `const_term` and `is_const_term` performing checks but returning different values. Can we maybe call this `is_model_term` or `is_value_term` instead?

Reply:
Sure